eventuals/concurrent.h

```diff
@@ -555,12 +554,13 @@ struct _Concurrent final {
                          .start([this](auto& k) {
                            if (exception_ && upstream_done_ && fibers_done_) {
                              // TODO(benh): flush remaining values first?
-                             try {
-                               std::rethrow_exception(*exception_);
-                             } catch (const StoppedException&) {
+                             if constexpr (!std::is_same_v<std::decay_t<decltype(exception_.value())>, Stopped>) {
+                               std::visit([this, &k](auto&& error) {
+                                 k.Fail(std::forward<decltype(error)>(error));
+                               },
+                                          exception_.value());
+                             } else {
                                k.Stop();
```

Comment:
I'm not sure the code you have will do what you want ... basically you're asking for the type of the `stopped_or_error_` at run-time but then trying to decide what you do with it at compile-time? I think we need to get the run-time value and add in checks at compile-time that if that run-time value is `Stopped` then do something different:
```suggestion
                            std::visit(
                                 [&k](auto&& error) {
                                   if constexpr (!std::is_same_v<std::decay_t<decltype(error)>, Stopped>) {
                                     k.Fail(std::forward<decltype(error)>(error));
                                   } else {
                                     k.Stop();
                                   }
                                 },
                                 stopped_or_value_->value());
```